{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.77\n"
     ]
    }
   ],
   "source": [
    "import Bio\n",
    "print(Bio.__version__)\n",
    "import os\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.Alphabet import IUPAC\n",
    "from Bio.Alphabet import generic_nucleotide\n",
    "from Bio import Align\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import MutableSeq\n",
    "from Bio.Alphabet import generic_dna\n",
    "import re\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment\n",
    "\n",
    "offsetapplicablepair = 10000000\n",
    "samenesslookahead = 2\n",
    "variantnumber = 10\n",
    "varheadoffset = 0\n",
    "var_seq_record = 0\n",
    "ref_gene_name = \"BRCA1\"\n",
    "ref_seq_record = SeqIO.read(ref_gene_name+\"-REF.fasta\", \"fasta\")\n",
    "\n",
    "dirpath = 'C:\\\\Users\\\\anjan.DESKTOP-PC5VIV8\\\\GPOC\\\\testfiles\\\\'\n",
    "\n",
    "testfilecount = 0\n",
    "testfilenames = []\n",
    "#reporting section\n",
    "reportlines = []\n",
    "spcr = \"|->| \"\n",
    "#strdiagnone = \"Diagnosis: None\"\n",
    "strdiagnone = \"---\"\n",
    "\n",
    "lstvarhits = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "reportlines.append(\"KAIGENIX CLINICAL MANAGEMENT - GENOMICS DEPT.\".center(100))\n",
    "strgeneline = \"Genetic Variations Report (for \"+ref_gene_name+\" gene)\\n\"\n",
    "reportlines.append(strgeneline.center(100))\n",
    "\n",
    "reportlines.append(\"Details:\")\n",
    "reportlines.append(\"--------------------------------------------------------------------------------------------------------------------------\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "for root, dirs, files in os.walk(dirpath):\n",
    "    for filename in files:\n",
    "            print(filename)\n",
    "            testfilecount+=1\n",
    "            testfilenames.append(filename)\n",
    "\n",
    "print(testfilecount)\n",
    "for i in range(testfilecount):\n",
    "    print(str(i) + \" -- \" + testfilenames[i])\n",
    "'''\n",
    "\n",
    "def getnetoffsetapplicable(start4digitnumber):\n",
    "    netoffset = 0\n",
    "    objectcount = len(lstvarhits)\n",
    "    print(\" in getnetoffsetapplicable function\")\n",
    "    for i in range(objectcount):\n",
    "        \n",
    "        tupobj = lstvarhits[i]\n",
    "        print(\"***** next tuple *****\")\n",
    "        print(\"tupobj[0]: \" + str(tupobj[0]))\n",
    "        print(\"tupobj[1]: \" + str(tupobj[1]))\n",
    "        if start4digitnumber > tupobj[0]:\n",
    "            netoffset += tupobj[1]\n",
    "            print(\"netoffset temp = \", netoffset)\n",
    "    \n",
    "    print(\"netoffset total = \", netoffset)   \n",
    "    return netoffset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nfor seq_record in SeqIO.parse(\"BRCA1-REF.fasta\", \"fasta\"):\\n    print(seq_record.id)\\n    print(repr(seq_record.seq))\\n    print(len(seq_record))\\n'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "for seq_record in SeqIO.parse(\"BRCA1-REF.fasta\", \"fasta\"):\n",
    "    print(seq_record.id)\n",
    "    print(repr(seq_record.seq))\n",
    "    print(len(seq_record))\n",
    "'''\n",
    "   \n",
    "#ref_seq_record = SeqIO.read(\"BRCA1-REF.fasta\", \"fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nif variantnumber == 1:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR01.fasta\", \"fasta\")\\nif variantnumber == 2:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR02.fasta\", \"fasta\")\\nif variantnumber == 3:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR03.fasta\", \"fasta\")\\nif variantnumber == 4:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR04.fasta\", \"fasta\")\\nif variantnumber == 5:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR05.fasta\", \"fasta\")\\nif variantnumber == 6:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR06.fasta\", \"fasta\")\\nif variantnumber == 7:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR07.fasta\", \"fasta\")\\nif variantnumber == 8:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR08.fasta\", \"fasta\")\\nif variantnumber == 9:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR09.fasta\", \"fasta\")\\nif variantnumber == 10:\\n    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR10.fasta\", \"fasta\") \\n'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "if variantnumber == 1:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR01.fasta\", \"fasta\")\n",
    "if variantnumber == 2:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR02.fasta\", \"fasta\")\n",
    "if variantnumber == 3:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR03.fasta\", \"fasta\")\n",
    "if variantnumber == 4:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR04.fasta\", \"fasta\")\n",
    "if variantnumber == 5:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR05.fasta\", \"fasta\")\n",
    "if variantnumber == 6:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR06.fasta\", \"fasta\")\n",
    "if variantnumber == 7:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR07.fasta\", \"fasta\")\n",
    "if variantnumber == 8:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR08.fasta\", \"fasta\")\n",
    "if variantnumber == 9:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR09.fasta\", \"fasta\")\n",
    "if variantnumber == 10:\n",
    "    var_seq_record = SeqIO.read(\"P001-BRCA1-VAR10.fasta\", \"fasta\") \n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nprint(var_seq_record.seq[:100])\\nprint(ref_seq_record.seq[:100])\\n'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "print(var_seq_record.seq[:100])\n",
    "print(ref_seq_record.seq[:100])\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def longform(strdiagnosis):\n",
    "    if strdiagnosis == \"BB\": return \"BENIGN\"\n",
    "    if strdiagnosis == \"LB\": return \"LIKELY BENIGN\"\n",
    "    if strdiagnosis == \"PP\": return \"PATHOLOGICAL\"\n",
    "    if strdiagnosis == \"LP\": return \"LIKELY PATHOLOGICAL\"\n",
    "    if strdiagnosis == \"NN\": return \"NOT KNOWN\"\n",
    "    return \"NOT KNOWN\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  sample -- c.0010C>T##BB \n",
    "def verifyvarient1present(variantrule):\n",
    "    print(\"in rule verification 1----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    strbeforesubstitution=variantrule[6]\n",
    "    straftersubstitution=variantrule[8]\n",
    "    print(strbeforesubstitution)\n",
    "    print(straftersubstitution)\n",
    "    print('substitution confirmation 1')\n",
    "    strdiagnosis = variantrule[11] + variantrule[12]\n",
    "    print(strdiagnosis)\n",
    "    print('var head offset 1--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    print(var_seq_record.seq[start4digitnumber + varheadoffsetTemp])\n",
    "    print(ref_seq_record.seq[start4digitnumber])\n",
    "    print('substitution confirmation 2')\n",
    "    if var_seq_record.seq[start4digitnumber + varheadoffsetTemp] == straftersubstitution:\n",
    "        if ref_seq_record.seq[start4digitnumber] == strbeforesubstitution:\n",
    "            reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "            print(variantrule + \"      PRESENT                               ******************\")\n",
    "        else:\n",
    "            reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "        \n",
    "    # no change to varheadoffset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  sample -- c.0005_0008del##LB\n",
    "def verifyvarient2present(variantrule):\n",
    "    print(\"in rule verification 2----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    \n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    strdiagnosis = variantrule[16] + variantrule[17]\n",
    "    print(strdiagnosis)\n",
    "    print('var head offset 2--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    deleted = False\n",
    "    nextdigitssame = True\n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+i])\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+i]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+i]:\n",
    "            deleted = True\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    print('deletion comfirmation above, NOT same is good')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment)  \n",
    "    print('same segment confirmation above, same is good')\n",
    "\n",
    "    \n",
    "    if deleted == True and nextdigitssame == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                              ******************\")\n",
    "        '''\n",
    "        varheadoffset = varheadoffset - difflength\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        varoffset = 0 - difflength\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0009_0012delTCCT##PP\n",
    "def verifyvarient3present(variantrule):\n",
    "    print(\"in rule verification 3----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    \n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "        #get the string\n",
    "    strdel=''\n",
    "    sequencepointer = 14\n",
    "    while variantrule[sequencepointer] != '#':\n",
    "        strdel = strdel + variantrule[sequencepointer]\n",
    "        sequencepointer += 1\n",
    "    print(strdel)\n",
    "    \n",
    "    sequencepointer += 2\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    print('var head offset 3--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    deleted = False\n",
    "    nextdigitssame = True\n",
    "    rightseqdeleted = True\n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+i])\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+i]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+i]:\n",
    "            deleted = True\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    print('deletion comfirmation above, NOT same is good')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[start4digitnumber + varheadoffset +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment)    \n",
    "    print('same segment comfirmation above, same is good')\n",
    "    \n",
    "    for k in range(difflength):\n",
    "        print('strdel ' + strdel[k])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+k])\n",
    "        if strdel[k] != ref_seq_record.seq[start4digitnumber+k]:\n",
    "            rightseqdeleted = False\n",
    "    print('right seq deleted comfirmation above, same is good')\n",
    "    \n",
    "    if deleted == True and rightseqdeleted == True and nextdigitssame == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                ******************\")\n",
    "        '''\n",
    "        varheadoffset = varheadoffset - difflength\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        varoffset = 0 - difflength\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "        \n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0009_0012dup##LP\n",
    "def verifyvarient4present(variantrule):\n",
    "    print(\"in rule verification 4----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "\n",
    "        \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    ''' \n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    strdiagnosis = variantrule[16] + variantrule[17]\n",
    "    print(strdiagnosis)\n",
    "    print('var head offset 4--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    #varheadoffset = 412\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    duplicated = True\n",
    "    nextdigitssame = True\n",
    "   \n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + var_seq_record.seq[end4digitnumber+1 + varheadoffsetTemp +i])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+i])\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[end4digitnumber +1 + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+i]\n",
    "        if var_seq_record.seq[end4digitnumber+1 +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+i]:\n",
    "            duplicated = False\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    print('duplication comfirmation above, same is good')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[end4digitnumber+difflength+1 + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[end4digitnumber+1+ difflength + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[end4digitnumber+1+difflength +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment)  \n",
    "    print('same segment comfirmation above, same is good')\n",
    "    \n",
    "    \n",
    "    if duplicated == True and nextdigitssame == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                  ******************\")\n",
    "        '''\n",
    "        varheadoffset = varheadoffset + difflength\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        varoffset = 0 + difflength\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "        \n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0014_0017dupGACG##NN\n",
    "def verifyvarient5present(variantrule):\n",
    "    print(\"in rule verification 5----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "\n",
    "\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    strdup=''\n",
    "    sequencepointer = 14\n",
    "    while variantrule[sequencepointer] != '#':\n",
    "        strdup = strdup + variantrule[sequencepointer]\n",
    "        sequencepointer += 1\n",
    "    print(strdup)\n",
    "    sequencepointer += 2\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    \n",
    "    print('var head offset 5--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    duplicated = True\n",
    "    nextdigitssame = True\n",
    "    correctduplicateseq = True\n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print(\"var pos = \", str(end4digitnumber+1 + varheadoffsetTemp +i))\n",
    "        print('var ' + var_seq_record.seq[end4digitnumber+1 + varheadoffsetTemp +i])\n",
    "        print(\"ref pos = \", str(start4digitnumber+i))\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+i])\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[end4digitnumber +1 + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+i]\n",
    "        if var_seq_record.seq[end4digitnumber+1 +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+i]:\n",
    "            duplicated = False\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    print('duplication comfirmation above, same is good')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[end4digitnumber+difflength+1 + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[end4digitnumber+1+ difflength + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[end4digitnumber+1+difflength +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment) \n",
    "    print('same segment comfirmation above, same is good')\n",
    "    \n",
    "    for k in range(difflength):\n",
    "        print('strdup ' + strdup[k])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+k])\n",
    "        if strdup[k] != ref_seq_record.seq[start4digitnumber+k]:\n",
    "            correctduplicateseq = False\n",
    "    print('correct duplicate seq comfirmation above, same is good')\n",
    "            \n",
    "            \n",
    "    if duplicated == True and nextdigitssame == True and correctduplicateseq == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                    ******************\")\n",
    "        \n",
    "        '''\n",
    "        varheadoffset = varheadoffset + difflength\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        varoffset = 0 + difflength\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "        \n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    #varheadoffset = 512\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0012_0013insATATAT##BB\n",
    "def verifyvarient6present(variantrule):\n",
    "    print(\"in rule verification 6----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    print(\"||||||||varheadoffsetTemp =\"+ str(varheadoffsetTemp)) \n",
    "    \n",
    "    strins=''\n",
    "    sequencepointer = 14\n",
    "    while variantrule[sequencepointer] != '#':\n",
    "        strins = strins + variantrule[sequencepointer]\n",
    "        sequencepointer += 1\n",
    "    print(\"strins: \" + strins)\n",
    "    lengthstrins = len(strins)\n",
    "    sequencepointer += 2\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    \n",
    "    print('var head offset 6--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(\"difflength = \", difflength)\n",
    "    \n",
    "    nextdigitssame = True\n",
    "    correctinsertedseq = True\n",
    "    \n",
    "    '''\n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(lengthstrins):\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber+1 + varheadoffsetTemp +i])\n",
    "        print(start4digitnumber+1 + varheadoffsetTemp +i)\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+1+i])\n",
    "        print(start4digitnumber+1+i)\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[start4digitnumber +1 + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+1+i]\n",
    "        if var_seq_record.seq[start4digitnumber+1 +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+1+i]:\n",
    "            duplicated = False\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    \n",
    "    print('duplicated? why not inserted? comfirmation above, NOT same is good')\n",
    "    '''\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[end4digitnumber+lengthstrins + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[end4digitnumber+ lengthstrins + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+j]\n",
    "        if var_seq_record.seq[end4digitnumber+lengthstrins +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment)    \n",
    "    print('same segment comfirmation above, same is good')\n",
    "    \n",
    "    for k in range(lengthstrins):\n",
    "        print('strins ' + strins[k])\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber+varheadoffsetTemp+1+k])\n",
    "        if strins[k] != var_seq_record.seq[start4digitnumber+varheadoffsetTemp+1+k]:\n",
    "            correctinsertedseq = False\n",
    "    \n",
    "    print('correct inserted seq comfirmation above, same is good')\n",
    "            \n",
    "            \n",
    "    if correctinsertedseq == True and nextdigitssame == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                    ******************\")\n",
    "        \n",
    "        '''\n",
    "        varheadoffset = varheadoffset + lengthstrins\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        varoffset = 0 + lengthstrins\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "        \n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    #varheadoffset = 512\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0009_0010delinsGGAA##LP\n",
    "def verifyvarient7present(variantrule):\n",
    "    print(\"in rule verification 7----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    strdelins=''\n",
    "    sequencepointer = 17\n",
    "    while variantrule[sequencepointer] != '#':\n",
    "        strdelins = strdelins + variantrule[sequencepointer]\n",
    "        sequencepointer += 1\n",
    "    print(strdelins)\n",
    "    lengthstrdelins = len(strdelins)\n",
    "    \n",
    "    sequencepointer += 2\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    \n",
    "    print('var head offset 7--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    #varheadoffset = 712\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    deleted = False\n",
    "    inserted = True\n",
    "    nextdigitssame = True\n",
    "    correctinsertedseq = True\n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+i])\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+i]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+i]:\n",
    "            deleted = True\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    \n",
    "    print('deleted comfirmation above, not same is true')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[start4digitnumber+lengthstrdelins + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[start4digitnumber+ lengthstrdelins + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[start4digitnumber+lengthstrdelins +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment) \n",
    "    \n",
    "    print('same segment comfirmation above, same is good')\n",
    "    \n",
    "    for k in range(lengthstrdelins):\n",
    "        print('strdelins ' + strdelins[k])\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber+varheadoffsetTemp+k])\n",
    "        if strdelins[k] != var_seq_record.seq[start4digitnumber+varheadoffsetTemp+k]:\n",
    "            correctinsertedseq = False\n",
    "    \n",
    "    print('correct inserted seq comfirmation above, same is good')\n",
    "    \n",
    "    if deleted == True and nextdigitssame == True and correctinsertedseq == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                               ******************\")\n",
    "        \n",
    "        '''\n",
    "        varheadoffset = varheadoffset - difflength + lengthstrdelins\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        varoffset = 0 - difflength + lengthstrdelins\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "        \n",
    "        \n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    #varheadoffset = 512\n",
    "   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0009_0010delTCinsGGAA##LP\n",
    "def verifyvarient8present(variantrule):\n",
    "    print(\"in rule verification 8 ----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    strdel=''\n",
    "    sequencepointer = 14\n",
    "    char = variantrule[sequencepointer]\n",
    "    print(char)\n",
    "    while ((char != 'I') and (char != 'i')):\n",
    "        strdel = strdel + char\n",
    "        sequencepointer += 1\n",
    "        char = variantrule[sequencepointer]\n",
    "        print(char)\n",
    "    print(strdel)\n",
    "    lengthstrdel = len(strdel)\n",
    "    \n",
    "    strins=''\n",
    "    sequencepointer += 3\n",
    "    while variantrule[sequencepointer] != '#':\n",
    "        strins = strins + variantrule[sequencepointer]\n",
    "        sequencepointer += 1\n",
    "    print(strins)\n",
    "    lengthstrins = len(strins)\n",
    "    \n",
    "    sequencepointer += 2\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    \n",
    "    print('var head offset 8--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    #varheadoffset = 712\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    deleted = False\n",
    "    inserted = True\n",
    "    nextdigitssame = True\n",
    "    correctdeletedseq = True\n",
    "    correctinsertedseq = True\n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i])\n",
    "        print(start4digitnumber + varheadoffsetTemp +i)\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+i])\n",
    "        print(start4digitnumber+i)\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumber+i]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumber+i]:\n",
    "            deleted = True\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    \n",
    "    print('deleted comfirmation above. not same is good')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[start4digitnumber+lengthstrins + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[start4digitnumber+ lengthstrins + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[start4digitnumber+lengthstrins +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment) \n",
    "    \n",
    "    print('same segment comfirmation above, same is good')\n",
    "    \n",
    "    for k in range(lengthstrdel):\n",
    "        print('strdel ' + strdel[k])\n",
    "        print('ref ' + ref_seq_record.seq[start4digitnumber+k])\n",
    "        if strdel[k] != ref_seq_record.seq[start4digitnumber+k]:\n",
    "            correctdeletedseq = False\n",
    "            \n",
    "    print('correct deleted seq comfirmation above, same is good')\n",
    "    \n",
    "    for l in range(lengthstrins):\n",
    "        print('strins ' + strins[l])\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber+varheadoffsetTemp+l])\n",
    "        if strins[l] != var_seq_record.seq[start4digitnumber+varheadoffsetTemp+l]:\n",
    "            correctinsertedseq = False\n",
    "    print('correct inserted seq comfirmation above, same is good')\n",
    "    \n",
    "    if deleted == True and nextdigitssame == True and correctinsertedseq == True and correctdeletedseq == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                       ******************\")\n",
    "        \n",
    "        '''\n",
    "        varheadoffset = varheadoffset + difflength\n",
    "        print('new var head offset --' + str(varheadoffset))\n",
    "        offsetapplicablepair = min(offsetapplicablepair, start4digitnumber)\n",
    "        print('new offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "        '''\n",
    "        \n",
    "        varoffset = 0 - lengthstrdel + lengthstrins\n",
    "        tup = (start4digitnumber, varoffset)\n",
    "        lstvarhits.append(tup)\n",
    "        \n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr+ \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    #varheadoffset = 512\n",
    "   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0021_0039inv##PP\n",
    "def verifyvarient9present(variantrule):\n",
    "    #c.0021_0039inv##PP\n",
    "    print(\"in rule verification 9 ----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "    #feature_seq = example_parent[example_feature.location.start:example_feature.location.end].reverse_complement()\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "\n",
    "    sequencepointer = 16\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    \n",
    "    print('var head offset 9--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    \n",
    "    isinverted = True\n",
    "    \n",
    "    nextdigitssame = True\n",
    "\n",
    "    invertedseq = var_seq_record.seq[start4digitnumber+varheadoffsetTemp:end4digitnumber+varheadoffsetTemp+1]\n",
    "    \n",
    "    correspondingrefseq = ref_seq_record.seq[start4digitnumber:end4digitnumber+1]\n",
    "    \n",
    "    \n",
    "    print(invertedseq)\n",
    "    originalseqfromvarseq = invertedseq.reverse_complement()\n",
    "    print(originalseqfromvarseq)\n",
    "    \n",
    "    print(correspondingrefseq)\n",
    "    \n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + originalseqfromvarseq[i])\n",
    "        print('ref ' + correspondingrefseq[i])\n",
    "        if originalseqfromvarseq[i] != correspondingrefseq[i]:\n",
    "            isinverted = False\n",
    "    print('strvarsegment which is inverted ' + invertedseq)\n",
    "    print('strvarsegment which is inverted back ' + originalseqfromvarseq)\n",
    "    print('strrefsegment ' + correspondingrefseq)\n",
    "    \n",
    "    print('inversion reverse complement comfirmation above, same [2 and 3] is good')\n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[start4digitnumber+difflength + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[start4digitnumber+ difflength + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[start4digitnumber+difflength +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment) \n",
    "    print('same segment comfirmation above, same is good')\n",
    "             \n",
    "    if isinverted == True and nextdigitssame == True :\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT  !!!!!!!!!!!!!!  \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                      ******************\")\n",
    "        #varheadoffset = varheadoffset + difflength ---commented, no change in length of the sequence due to the variation\n",
    "        #varheadoffsetTemp = min(varheadoffsetTemp, start4digitnumber)\n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample -- c.0021_0039con0061_0079##LP\n",
    "def verifyvarient10present(variantrule):\n",
    "    print(\"in rule verification 10 ----------------------\" + variantrule)\n",
    "    global varheadoffset\n",
    "    global offsetapplicablepair\n",
    "    global var_seq_record\n",
    "    global ref_seq_record\n",
    "   \n",
    "    #c.0021_0039con0061_0079##LP\n",
    "    \n",
    "    start4digitnumber = int(variantrule[2])*1000 + int(variantrule[3])*100 + int(variantrule[4])*10 + int(variantrule[5])*1\n",
    "    print(start4digitnumber)\n",
    "    end4digitnumber = int(variantrule[7])*1000 + int(variantrule[8])*100 + int(variantrule[9])*10 + int(variantrule[10])*1\n",
    "    print(end4digitnumber)\n",
    "    \n",
    "    '''\n",
    "    if offsetapplicablepair < start4digitnumber:\n",
    "        varheadoffsetTemp = varheadoffset\n",
    "    else:\n",
    "        varheadoffsetTemp = 0\n",
    "    '''\n",
    "    varheadoffsetTemp = getnetoffsetapplicable(start4digitnumber)\n",
    "    \n",
    "    start4digitnumbercon = int(variantrule[14])*1000 + int(variantrule[14])*100 + int(variantrule[16])*10 + int(variantrule[17])*1\n",
    "    print(start4digitnumbercon)\n",
    "    end4digitnumbercon = int(variantrule[19])*1000 + int(variantrule[20])*100 + int(variantrule[21])*10 + int(variantrule[22])*1\n",
    "    print(end4digitnumbercon)\n",
    "\n",
    "    sequencepointer = 25\n",
    "    strdiagnosis = variantrule[sequencepointer] + variantrule[sequencepointer+1]\n",
    "    print(strdiagnosis)\n",
    "    \n",
    "    print('var head offset 10--' + str(varheadoffsetTemp))\n",
    "    print('offsetapplicablepair == ' + str(offsetapplicablepair))\n",
    "    \n",
    "    print(\"FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\")\n",
    "    \n",
    "    islengthsame = False\n",
    "    isconverted = True\n",
    "    nextdigitssame = True\n",
    "    \n",
    "    difflength = end4digitnumber - start4digitnumber +1\n",
    "    print(difflength)\n",
    "    difflengthcon = end4digitnumbercon - start4digitnumbercon +1\n",
    "    print(difflengthcon)\n",
    "    if difflength == difflengthcon:\n",
    "        islengthsame = True\n",
    "    \n",
    "    strvarsegment = \"\"\n",
    "    strrefsegment = \"\"\n",
    "    for i in range(difflength):\n",
    "        print('var ' + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i])\n",
    "        print('refcon ' + ref_seq_record.seq[start4digitnumbercon+i])\n",
    "        strvarsegment = strvarsegment + var_seq_record.seq[start4digitnumber + varheadoffsetTemp +i]\n",
    "        strrefsegment = strrefsegment + ref_seq_record.seq[start4digitnumbercon+i]\n",
    "        if var_seq_record.seq[start4digitnumber +varheadoffsetTemp +i] != ref_seq_record.seq[start4digitnumbercon+i]:\n",
    "            isconverted = False\n",
    "    print('strvarsegment ' + strvarsegment)\n",
    "    print('strrefsegment ' + strrefsegment)\n",
    "    print('converted comfirmation above, i.e same [var, ref-con] is good')\n",
    "    \n",
    "    \n",
    "    strnextvarsegment = \"\"\n",
    "    strnextrefsegment = \"\"\n",
    "    for j in range(samenesslookahead):\n",
    "        print('var next ' + var_seq_record.seq[start4digitnumber+difflength + varheadoffsetTemp +j])\n",
    "        print('ref next ' + ref_seq_record.seq[end4digitnumber+1+j])\n",
    "        strnextvarsegment = strnextvarsegment + var_seq_record.seq[start4digitnumber+ difflength + varheadoffsetTemp +j]\n",
    "        strnextrefsegment = strnextrefsegment + ref_seq_record.seq[end4digitnumber+1+j]\n",
    "        if var_seq_record.seq[start4digitnumber+difflength +varheadoffsetTemp +j] != ref_seq_record.seq[end4digitnumber+1+j]:\n",
    "            nextdigitssame = False\n",
    "    print('strnextvarsegment ' + strnextvarsegment)\n",
    "    print('strnextrefsegment ' + strnextrefsegment) \n",
    "    print('same segment comfirmation above, same is good')\n",
    "             \n",
    "    if isconverted == True and nextdigitssame == True and islengthsame == True:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"PRESENT   !!!!!!!!!!!!!! \".ljust(30)+ spcr + \"Diagnosis: \" + longform(strdiagnosis) )\n",
    "        print(variantrule + \"      PRESENT                                         ******************\")\n",
    "        #varheadoffset = varheadoffset + difflength ---commented, no change in length of the sequence due to the variation\n",
    "        #varheadoffsetTemp = min(varheadoffsetTemp, start4digitnumber)\n",
    "    else:\n",
    "        reportlines.append(\"VARIANT \"+ spcr + variantrule.ljust(40)+ spcr + \"Absent\".ljust(30)+ spcr + strdiagnone )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "def checkvariantexists(variantnumber, variantrule ):\n",
    "    #print(variantnumber)\n",
    "    #print(variantrule)\n",
    "    match = 0\n",
    "    matchObj1 = re.match( r'c\\.\\d\\d\\d\\d[a-zA-Z]>[a-zA-Z]##', variantrule)\n",
    "    if matchObj1:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 1')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient1present(variantrule)\n",
    "        \n",
    "    matchObj2 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Dd][Ee][Ll]##', variantrule)\n",
    "    if matchObj2:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 2')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient2present(variantrule)\n",
    "        \n",
    "    matchObj3 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Dd][Ee][Ll][atgcATGC]+##', variantrule)\n",
    "    if matchObj3:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 3')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient3present(variantrule)\n",
    "        \n",
    "    matchObj4 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Dd][Uu][Pp]##', variantrule)\n",
    "    if matchObj4:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 4')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient4present(variantrule)\n",
    "        \n",
    "    matchObj5 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Dd][Uu][Pp][atgcATGC]+##', variantrule)\n",
    "    if matchObj5:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 5')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient5present(variantrule)\n",
    "        \n",
    "    matchObj6 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Ii][Nn][Ss][atgcATGC]+##', variantrule)\n",
    "    if matchObj6:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 6')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient6present(variantrule)\n",
    "        \n",
    "    matchObj7 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Dd][Ee][Ll][Ii][Nn][Ss][atgcATGC]+##', variantrule)\n",
    "    if matchObj7:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 7')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient7present(variantrule)\n",
    "        \n",
    "    matchObj8 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Dd][Ee][Ll][atgcATGC]+[Ii][Nn][Ss][atgcATGC]+##', variantrule)\n",
    "    if matchObj8:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 8')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient8present(variantrule)\n",
    "    \n",
    "    #0021_0039inv##PP\n",
    "    matchObj9 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Ii][Nn][Vv]##', variantrule)\n",
    "    if matchObj9:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 9')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient9present(variantrule)\n",
    "        \n",
    "    matchObj10 = re.match( r'c\\.\\d\\d\\d\\d_\\d\\d\\d\\d[Cc][Oo][Nn]\\d\\d\\d\\d_\\d\\d\\d\\d##', variantrule)\n",
    "    if matchObj10:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- matched 10')\n",
    "        match = 1\n",
    "        isvarientpresent = verifyvarient10present(variantrule)\n",
    "        \n",
    "    if match == 0:\n",
    "        print('\\n\\n Variant#' + str(variantnumber)+' Variant:'+variantrule + ' --- NO MATCHING RULE FOUND!!!!!')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nprint(var_seq_record.seq[:100])\\nprint(ref_seq_record.seq[:100])\\n'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "print(var_seq_record.seq[:100])\n",
    "print(ref_seq_record.seq[:100])\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P004-BRCA1-CV01.fasta\n",
      "P004-BRCA1-CV02.fasta\n",
      "2\n",
      "0 -- P004-BRCA1-CV01.fasta\n",
      "\n",
      "\n",
      " Variant#1 Variant:c.0009_0012del##BB --- matched 2\n",
      "in rule verification 2----------------------c.0009_0012del##BB\n",
      "9\n",
      "12\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "BB\n",
      "var head offset 2--0\n",
      "offsetapplicablepair == 10000000\n",
      "4\n",
      "var T\n",
      "ref T\n",
      "var G\n",
      "ref C\n",
      "var C\n",
      "ref C\n",
      "var T\n",
      "ref T\n",
      "strvarsegment TGCT\n",
      "strrefsegment TCCT\n",
      "deletion comfirmation above, NOT same is good\n",
      "var next T\n",
      "ref next G\n",
      "var next G\n",
      "ref next G\n",
      "strnextvarsegment TG\n",
      "strnextrefsegment GG\n",
      "same segment confirmation above, same is good\n",
      "\n",
      "\n",
      " Variant#2 Variant:c.0009_0012deqlTCCT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#3 Variant:c.0010C>G##LB --- matched 1\n",
      "in rule verification 1----------------------c.0010C>G##LB\n",
      "10\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "C\n",
      "G\n",
      "substitution confirmation 1\n",
      "LB\n",
      "var head offset 1--0\n",
      "offsetapplicablepair == 10000000\n",
      "G\n",
      "C\n",
      "substitution confirmation 2\n",
      "c.0010C>G##LB      PRESENT                               ******************\n",
      "\n",
      "\n",
      " Variant#4 Variant:c.0021_0030inv##LP --- matched 9\n",
      "in rule verification 9 ----------------------c.0021_0030inv##LP\n",
      "21\n",
      "30\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "LP\n",
      "var head offset 9--0\n",
      "offsetapplicablepair == 10000000\n",
      "10\n",
      "ACAGCCTGTC\n",
      "GACAGGCTGT\n",
      "GACAGGCTGT\n",
      "var G\n",
      "ref G\n",
      "var A\n",
      "ref A\n",
      "var C\n",
      "ref C\n",
      "var A\n",
      "ref A\n",
      "var G\n",
      "ref G\n",
      "var G\n",
      "ref G\n",
      "var C\n",
      "ref C\n",
      "var T\n",
      "ref T\n",
      "var G\n",
      "ref G\n",
      "var T\n",
      "ref T\n",
      "strvarsegment which is inverted ACAGCCTGTC\n",
      "strvarsegment which is inverted back GACAGGCTGT\n",
      "strrefsegment GACAGGCTGT\n",
      "inversion reverse complement comfirmation above, same [2 and 3] is good\n",
      "var next G\n",
      "ref next G\n",
      "var next G\n",
      "ref next G\n",
      "strnextvarsegment GG\n",
      "strnextrefsegment GG\n",
      "same segment comfirmation above, same is good\n",
      "c.0021_0030inv##LP      PRESENT                                      ******************\n",
      "\n",
      "\n",
      " Variant#5 Variant:c.0022_0023insCTAG##NN --- matched 6\n",
      "in rule verification 6----------------------c.0022_0023insCTAG##NN\n",
      "22\n",
      "23\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "||||||||varheadoffsetTemp =0\n",
      "strins: CTAG\n",
      "NN\n",
      "var head offset 6--0\n",
      "offsetapplicablepair == 10000000\n",
      "difflength =  2\n",
      "var next T\n",
      "ref next C\n",
      "var next G\n",
      "ref next A\n",
      "strnextvarsegment TG\n",
      "strnextrefsegment CA\n",
      "same segment comfirmation above, same is good\n",
      "strins C\n",
      "var A\n",
      "strins T\n",
      "var G\n",
      "strins A\n",
      "var C\n",
      "strins G\n",
      "var C\n",
      "correct inserted seq comfirmation above, same is good\n",
      "\n",
      "\n",
      " Variant#6 Variant:c.0025_0028dup##PP --- matched 4\n",
      "in rule verification 4----------------------c.0025_0028dup##PP\n",
      "25\n",
      "28\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "PP\n",
      "var head offset 4--0\n",
      "offsetapplicablepair == 10000000\n",
      "4\n",
      "var T\n",
      "ref G\n",
      "var C\n",
      "ref G\n",
      "var G\n",
      "ref C\n",
      "var G\n",
      "ref T\n",
      "strvarsegment TCGG\n",
      "strrefsegment GGCT\n",
      "duplication comfirmation above, same is good\n",
      "var next G\n",
      "ref next G\n",
      "var next G\n",
      "ref next T\n",
      "strnextvarsegment GG\n",
      "strnextrefsegment GT\n",
      "same segment comfirmation above, same is good\n",
      "\n",
      "\n",
      " Variant#7 Variant:c.0025_0028duqpGGCT##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#8 Variant:c.0037_0038delinsCAAT##LP --- matched 7\n",
      "in rule verification 7----------------------c.0037_0038delinsCAAT##LP\n",
      "37\n",
      "38\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "CAAT\n",
      "LP\n",
      "var head offset 7--0\n",
      "offsetapplicablepair == 10000000\n",
      "2\n",
      "var T\n",
      "ref T\n",
      "var C\n",
      "ref C\n",
      "strvarsegment TC\n",
      "strrefsegment TC\n",
      "deleted comfirmation above, not same is true\n",
      "var next T\n",
      "ref next T\n",
      "var next C\n",
      "ref next C\n",
      "strnextvarsegment TC\n",
      "strnextrefsegment TC\n",
      "same segment comfirmation above, same is good\n",
      "strdelins C\n",
      "var T\n",
      "strdelins A\n",
      "var C\n",
      "strdelins A\n",
      "var T\n",
      "strdelins T\n",
      "var C\n",
      "correct inserted seq comfirmation above, same is good\n",
      "\n",
      "\n",
      " Variant#9 Variant:c.0037_0038deqlTCinsCAAT##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#10 Variant:c.0041_0050con0061_0070##PP --- matched 10\n",
      "in rule verification 10 ----------------------c.0041_0050con0061_0070##PP\n",
      "41\n",
      "50\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "61\n",
      "70\n",
      "PP\n",
      "var head offset 10--0\n",
      "offsetapplicablepair == 10000000\n",
      "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "10\n",
      "10\n",
      "var T\n",
      "refcon T\n",
      "var C\n",
      "refcon C\n",
      "var A\n",
      "refcon A\n",
      "var G\n",
      "refcon G\n",
      "var G\n",
      "refcon G\n",
      "var A\n",
      "refcon A\n",
      "var G\n",
      "refcon G\n",
      "var G\n",
      "refcon G\n",
      "var C\n",
      "refcon C\n",
      "var C\n",
      "refcon C\n",
      "strvarsegment TCAGGAGGCC\n",
      "strrefsegment TCAGGAGGCC\n",
      "converted comfirmation above, i.e same [var, ref-con] is good\n",
      "var next G\n",
      "ref next G\n",
      "var next C\n",
      "ref next C\n",
      "strnextvarsegment GC\n",
      "strnextrefsegment GC\n",
      "same segment comfirmation above, same is good\n",
      "c.0041_0050con0061_0070##PP      PRESENT                                         ******************\n",
      "\n",
      "\n",
      " Variant#11 Variant:--------------------------------------- --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#12 Variant:c.0010Cq>T##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#13 Variant:c.0005_0008dqel##LB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#14 Variant:c.0009_0012dqelTCCT##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#15 Variant:c.0009_0012dqup##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#16 Variant:c.0014_0017dqupGACG##NN --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#17 Variant:c.0012_0013inqsATATAT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#18 Variant:c.0009_0010deqlinsGGAA##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#19 Variant:c.0009_0010deqlTCinsTAGC##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#20 Variant:c.0021_0039inqv##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#21 Variant:c.0021_0039coqn0061_0079##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#22 Variant:c.0020Gq>T##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#23 Variant:c.0015_0018deql##LB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#24 Variant:c.0019_0022deqlGGGA##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#25 Variant:c.0019_0022duqp##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#26 Variant:c.0023_0026duqpCAGG##NN --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#27 Variant:c.0022_0023inqsATATAT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#28 Variant:c.0020_0021deqlinsCATG##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#29 Variant:c.0019_0020deqlGGinsCAGT##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#30 Variant:c.0031_0049inqv##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#31 Variant:c.0031_0049coqn0071_0089##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#32 Variant:c.0030Tq>A##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#33 Variant:c.0025_0028deql##LB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#34 Variant:c.0030_0033deqlTGGG##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#35 Variant:c.0029_0032duqp##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#36 Variant:c.0040_0043duqpCAGA##NN --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#37 Variant:c.0032_0033inqsATATAT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#38 Variant:c.0029_0030deqlinsGGAA##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#39 Variant:c.0029_0030deqlGTinsCCTT##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#40 Variant:c.0041_0059inqv##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#41 Variant:c.0041_0059coqn0081_0099##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "1 -- P004-BRCA1-CV02.fasta\n",
      "\n",
      "\n",
      " Variant#1 Variant:c.0009_0012del##BB --- matched 2\n",
      "in rule verification 2----------------------c.0009_0012del##BB\n",
      "9\n",
      "12\n",
      " in getnetoffsetapplicable function\n",
      "netoffset total =  0\n",
      "BB\n",
      "var head offset 2--0\n",
      "offsetapplicablepair == 10000000\n",
      "4\n",
      "var G\n",
      "ref T\n",
      "var G\n",
      "ref C\n",
      "var A\n",
      "ref C\n",
      "var C\n",
      "ref T\n",
      "strvarsegment GGAC\n",
      "strrefsegment TCCT\n",
      "deletion comfirmation above, NOT same is good\n",
      "var next G\n",
      "ref next G\n",
      "var next G\n",
      "ref next G\n",
      "strnextvarsegment GG\n",
      "strnextrefsegment GG\n",
      "same segment confirmation above, same is good\n",
      "c.0009_0012del##BB      PRESENT                              ******************\n",
      "\n",
      "\n",
      " Variant#2 Variant:c.0009_0012deqlTCCT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#3 Variant:c.0010C>G##LB --- matched 1\n",
      "in rule verification 1----------------------c.0010C>G##LB\n",
      "10\n",
      " in getnetoffsetapplicable function\n",
      "***** next tuple *****\n",
      "tupobj[0]: 9\n",
      "tupobj[1]: -4\n",
      "netoffset temp =  -4\n",
      "netoffset total =  -4\n",
      "C\n",
      "G\n",
      "substitution confirmation 1\n",
      "LB\n",
      "var head offset 1---4\n",
      "offsetapplicablepair == 10000000\n",
      "A\n",
      "C\n",
      "substitution confirmation 2\n",
      "\n",
      "\n",
      " Variant#4 Variant:c.0021_0030inv##LP --- matched 9\n",
      "in rule verification 9 ----------------------c.0021_0030inv##LP\n",
      "21\n",
      "30\n",
      " in getnetoffsetapplicable function\n",
      "***** next tuple *****\n",
      "tupobj[0]: 9\n",
      "tupobj[1]: -4\n",
      "netoffset temp =  -4\n",
      "netoffset total =  -4\n",
      "LP\n",
      "var head offset 9---4\n",
      "offsetapplicablepair == 10000000\n",
      "10\n",
      "GACTAGCAGG\n",
      "CCTGCTAGTC\n",
      "GACAGGCTGT\n",
      "var C\n",
      "ref G\n",
      "var C\n",
      "ref A\n",
      "var T\n",
      "ref C\n",
      "var G\n",
      "ref A\n",
      "var C\n",
      "ref G\n",
      "var T\n",
      "ref G\n",
      "var A\n",
      "ref C\n",
      "var G\n",
      "ref T\n",
      "var T\n",
      "ref G\n",
      "var C\n",
      "ref T\n",
      "strvarsegment which is inverted GACTAGCAGG\n",
      "strvarsegment which is inverted back CCTGCTAGTC\n",
      "strrefsegment GACAGGCTGT\n",
      "inversion reverse complement comfirmation above, same [2 and 3] is good\n",
      "var next C\n",
      "ref next G\n",
      "var next T\n",
      "ref next G\n",
      "strnextvarsegment CT\n",
      "strnextrefsegment GG\n",
      "same segment comfirmation above, same is good\n",
      "\n",
      "\n",
      " Variant#5 Variant:c.0022_0023insCTAG##NN --- matched 6\n",
      "in rule verification 6----------------------c.0022_0023insCTAG##NN\n",
      "22\n",
      "23\n",
      " in getnetoffsetapplicable function\n",
      "***** next tuple *****\n",
      "tupobj[0]: 9\n",
      "tupobj[1]: -4\n",
      "netoffset temp =  -4\n",
      "netoffset total =  -4\n",
      "||||||||varheadoffsetTemp =-4\n",
      "strins: CTAG\n",
      "NN\n",
      "var head offset 6---4\n",
      "offsetapplicablepair == 10000000\n",
      "difflength =  2\n",
      "var next C\n",
      "ref next C\n",
      "var next A\n",
      "ref next A\n",
      "strnextvarsegment CA\n",
      "strnextrefsegment CA\n",
      "same segment comfirmation above, same is good\n",
      "strins C\n",
      "var C\n",
      "strins T\n",
      "var T\n",
      "strins A\n",
      "var A\n",
      "strins G\n",
      "var G\n",
      "correct inserted seq comfirmation above, same is good\n",
      "c.0022_0023insCTAG##NN      PRESENT                                    ******************\n",
      "\n",
      "\n",
      " Variant#6 Variant:c.0025_0028dup##PP --- matched 4\n",
      "in rule verification 4----------------------c.0025_0028dup##PP\n",
      "25\n",
      "28\n",
      " in getnetoffsetapplicable function\n",
      "***** next tuple *****\n",
      "tupobj[0]: 9\n",
      "tupobj[1]: -4\n",
      "netoffset temp =  -4\n",
      "***** next tuple *****\n",
      "tupobj[0]: 22\n",
      "tupobj[1]: 4\n",
      "netoffset temp =  0\n",
      "netoffset total =  0\n",
      "PP\n",
      "var head offset 4--0\n",
      "offsetapplicablepair == 10000000\n",
      "4\n",
      "var G\n",
      "ref G\n",
      "var G\n",
      "ref G\n",
      "var C\n",
      "ref C\n",
      "var T\n",
      "ref T\n",
      "strvarsegment GGCT\n",
      "strrefsegment GGCT\n",
      "duplication comfirmation above, same is good\n",
      "var next G\n",
      "ref next G\n",
      "var next T\n",
      "ref next T\n",
      "strnextvarsegment GT\n",
      "strnextrefsegment GT\n",
      "same segment comfirmation above, same is good\n",
      "c.0025_0028dup##PP      PRESENT                                  ******************\n",
      "\n",
      "\n",
      " Variant#7 Variant:c.0025_0028duqpGGCT##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#8 Variant:c.0037_0038delinsCAAT##LP --- matched 7\n",
      "in rule verification 7----------------------c.0037_0038delinsCAAT##LP\n",
      "37\n",
      "38\n",
      " in getnetoffsetapplicable function\n",
      "***** next tuple *****\n",
      "tupobj[0]: 9\n",
      "tupobj[1]: -4\n",
      "netoffset temp =  -4\n",
      "***** next tuple *****\n",
      "tupobj[0]: 22\n",
      "tupobj[1]: 4\n",
      "netoffset temp =  0\n",
      "***** next tuple *****\n",
      "tupobj[0]: 25\n",
      "tupobj[1]: 4\n",
      "netoffset temp =  4\n",
      "netoffset total =  4\n",
      "CAAT\n",
      "LP\n",
      "var head offset 7--4\n",
      "offsetapplicablepair == 10000000\n",
      "2\n",
      "var C\n",
      "ref T\n",
      "var A\n",
      "ref C\n",
      "strvarsegment CA\n",
      "strrefsegment TC\n",
      "deleted comfirmation above, not same is true\n",
      "var next T\n",
      "ref next T\n",
      "var next C\n",
      "ref next C\n",
      "strnextvarsegment TC\n",
      "strnextrefsegment TC\n",
      "same segment comfirmation above, same is good\n",
      "strdelins C\n",
      "var C\n",
      "strdelins A\n",
      "var A\n",
      "strdelins A\n",
      "var A\n",
      "strdelins T\n",
      "var T\n",
      "correct inserted seq comfirmation above, same is good\n",
      "c.0037_0038delinsCAAT##LP      PRESENT                                               ******************\n",
      "\n",
      "\n",
      " Variant#9 Variant:c.0037_0038deqlTCinsCAAT##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#10 Variant:c.0041_0050con0061_0070##PP --- matched 10\n",
      "in rule verification 10 ----------------------c.0041_0050con0061_0070##PP\n",
      "41\n",
      "50\n",
      " in getnetoffsetapplicable function\n",
      "***** next tuple *****\n",
      "tupobj[0]: 9\n",
      "tupobj[1]: -4\n",
      "netoffset temp =  -4\n",
      "***** next tuple *****\n",
      "tupobj[0]: 22\n",
      "tupobj[1]: 4\n",
      "netoffset temp =  0\n",
      "***** next tuple *****\n",
      "tupobj[0]: 25\n",
      "tupobj[1]: 4\n",
      "netoffset temp =  4\n",
      "***** next tuple *****\n",
      "tupobj[0]: 37\n",
      "tupobj[1]: 2\n",
      "netoffset temp =  6\n",
      "netoffset total =  6\n",
      "61\n",
      "70\n",
      "PP\n",
      "var head offset 10--6\n",
      "offsetapplicablepair == 10000000\n",
      "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "10\n",
      "10\n",
      "var A\n",
      "refcon T\n",
      "var G\n",
      "refcon C\n",
      "var A\n",
      "refcon A\n",
      "var T\n",
      "refcon G\n",
      "var A\n",
      "refcon G\n",
      "var A\n",
      "refcon A\n",
      "var C\n",
      "refcon G\n",
      "var T\n",
      "refcon G\n",
      "var G\n",
      "refcon C\n",
      "var G\n",
      "refcon C\n",
      "strvarsegment AGATAACTGG\n",
      "strrefsegment TCAGGAGGCC\n",
      "converted comfirmation above, i.e same [var, ref-con] is good\n",
      "var next G\n",
      "ref next G\n",
      "var next C\n",
      "ref next C\n",
      "strnextvarsegment GC\n",
      "strnextrefsegment GC\n",
      "same segment comfirmation above, same is good\n",
      "\n",
      "\n",
      " Variant#11 Variant:--------------------------------------- --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#12 Variant:c.0010Cq>T##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#13 Variant:c.0005_0008dqel##LB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#14 Variant:c.0009_0012dqelTCCT##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#15 Variant:c.0009_0012dqup##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#16 Variant:c.0014_0017dqupGACG##NN --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#17 Variant:c.0012_0013inqsATATAT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#18 Variant:c.0009_0010deqlinsGGAA##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#19 Variant:c.0009_0010deqlTCinsTAGC##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#20 Variant:c.0021_0039inqv##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#21 Variant:c.0021_0039coqn0061_0079##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#22 Variant:c.0020Gq>T##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#23 Variant:c.0015_0018deql##LB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#24 Variant:c.0019_0022deqlGGGA##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#25 Variant:c.0019_0022duqp##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#26 Variant:c.0023_0026duqpCAGG##NN --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#27 Variant:c.0022_0023inqsATATAT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#28 Variant:c.0020_0021deqlinsCATG##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#29 Variant:c.0019_0020deqlGGinsCAGT##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#30 Variant:c.0031_0049inqv##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#31 Variant:c.0031_0049coqn0071_0089##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#32 Variant:c.0030Tq>A##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#33 Variant:c.0025_0028deql##LB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#34 Variant:c.0030_0033deqlTGGG##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#35 Variant:c.0029_0032duqp##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#36 Variant:c.0040_0043duqpCAGA##NN --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#37 Variant:c.0032_0033inqsATATAT##BB --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#38 Variant:c.0029_0030deqlinsGGAA##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#39 Variant:c.0029_0030deqlGTinsCCTT##LP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#40 Variant:c.0041_0059inqv##PP --- NO MATCHING RULE FOUND!!!!!\n",
      "\n",
      "\n",
      " Variant#41 Variant:c.0041_0059coqn0081_0099##LP --- NO MATCHING RULE FOUND!!!!!\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "netdeletes=0\n",
    "netinserts=0\n",
    "netduplicates=0\n",
    "\n",
    "global var_seq_record\n",
    "global ref_seq_record\n",
    "\n",
    "file1 = open('BRCA1-VARIANTS.txt', 'r') \n",
    "Variants = file1.readlines() \n",
    "\n",
    "for root, dirs, files in os.walk(dirpath):\n",
    "    for filename in files:\n",
    "            print(filename)\n",
    "            testfilecount+=1\n",
    "            testfilenames.append(filename)\n",
    "\n",
    "print(testfilecount)\n",
    "for i in range(testfilecount):\n",
    "    print(str(i) + \" -- \" + testfilenames[i])\n",
    "    var_seq_record = SeqIO.read(dirpath+testfilenames[i], \"fasta\")\n",
    "    reportlines.append(\"--------------------------------------------------------------------------------------------------------------------------\")\n",
    "    reportlines.append(\"For file:   \" + testfilenames[i])\n",
    "    reportlines.append(var_seq_record.seq[:20]+'|'+var_seq_record.seq[20:40]+'|'+var_seq_record.seq[40:60]+'|'+var_seq_record.seq[60:80]+'|'+var_seq_record.seq[80:100])\n",
    "    reportlines.append(ref_seq_record.seq[:20]+'|'+ref_seq_record.seq[20:40]+'|'+ref_seq_record.seq[40:60]+'|'+ref_seq_record.seq[60:80]+'|'+ref_seq_record.seq[80:100])\n",
    "    reportlines.append(\"--------------------------------------------------------------------------------------------------------------------------\")\n",
    "    varheadoffset = 0\n",
    "    lstvarhits = []\n",
    "    count = 0\n",
    "    # Strips the newline character \n",
    "    for variant in Variants: \n",
    "        #print(\"Line {}: {}\".format(count+1, variant.strip()))\n",
    "        checkvariantexists(count+1, variant.strip() )\n",
    "        count+=1\n",
    "        \n",
    "reportlines.append(\"--------------------------------------------------------------------------------------------------------------------------\")\n",
    "reportlines.append(\"===============================================END OF GENOMICS REPORT=====================================================\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                           KAIGENIX CLINICAL MANAGEMENT - GENOMICS DEPT.                            \n",
      "                            Genetic Variations Report (for BRCA1 gene)\n",
      "                             \n",
      "Details:\n",
      "--------------------------------------------------------------------------------------------------------------------------\n",
      "--------------------------------------------------------------------------------------------------------------------------\n",
      "For file:   P004-BRCA1-CV01.fasta\n",
      "GCTGAGACTTGCTGGACGGG|GACAGCCTGTCGGGGTTTCT|CTCAGGAGGCCGCCCCTGCG|CTCAGGAGGCCTTCACCCTC|TGCTCTGGGTAAAGTTCATT\n",
      "GCTGAGACTTCCTGGACGGG|GGACAGGCTGTGGGGTTTCT|CAGATAACTGGGCCCCTGCG|CTCAGGAGGCCTTCACCCTC|TGCTCTGGGTAAAGTTCATT\n",
      "--------------------------------------------------------------------------------------------------------------------------\n",
      "VARIANT |->| c.0009_0012del##BB                      |->| Absent                        |->| ---\n",
      "VARIANT |->| c.0010C>G##LB                           |->| PRESENT  !!!!!!!!!!!!!!       |->| Diagnosis: LIKELY BENIGN\n",
      "VARIANT |->| c.0021_0030inv##LP                      |->| PRESENT  !!!!!!!!!!!!!!       |->| Diagnosis: LIKELY PATHOLOGICAL\n",
      "VARIANT |->| c.0022_0023insCTAG##NN                  |->| Absent                        |->| ---\n",
      "VARIANT |->| c.0025_0028dup##PP                      |->| Absent                        |->| ---\n",
      "VARIANT |->| c.0037_0038delinsCAAT##LP               |->| Absent                        |->| ---\n",
      "VARIANT |->| c.0041_0050con0061_0070##PP             |->| PRESENT   !!!!!!!!!!!!!!      |->| Diagnosis: PATHOLOGICAL\n",
      "--------------------------------------------------------------------------------------------------------------------------\n",
      "For file:   P004-BRCA1-CV02.fasta\n",
      "GCTGAGACTGGACGGGGGAC|TAGCAGGCTGGCTGTGGGGT|TCAATTCAGATAACTGGGCC|CCTGCGCTCAGGAGGCCTTC|ACCCTCTGCTCTGGGTAAAG\n",
      "GCTGAGACTTCCTGGACGGG|GGACAGGCTGTGGGGTTTCT|CAGATAACTGGGCCCCTGCG|CTCAGGAGGCCTTCACCCTC|TGCTCTGGGTAAAGTTCATT\n",
      "--------------------------------------------------------------------------------------------------------------------------\n",
      "VARIANT |->| c.0009_0012del##BB                      |->| PRESENT  !!!!!!!!!!!!!!       |->| Diagnosis: BENIGN\n",
      "VARIANT |->| c.0010C>G##LB                           |->| Absent                        |->| ---\n",
      "VARIANT |->| c.0021_0030inv##LP                      |->| Absent                        |->| ---\n",
      "VARIANT |->| c.0022_0023insCTAG##NN                  |->| PRESENT  !!!!!!!!!!!!!!       |->| Diagnosis: NOT KNOWN\n",
      "VARIANT |->| c.0025_0028dup##PP                      |->| PRESENT  !!!!!!!!!!!!!!       |->| Diagnosis: PATHOLOGICAL\n",
      "VARIANT |->| c.0037_0038delinsCAAT##LP               |->| PRESENT  !!!!!!!!!!!!!!       |->| Diagnosis: LIKELY PATHOLOGICAL\n",
      "VARIANT |->| c.0041_0050con0061_0070##PP             |->| Absent                        |->| ---\n",
      "--------------------------------------------------------------------------------------------------------------------------\n",
      "===============================================END OF GENOMICS REPORT=====================================================\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(reportlines)):\n",
    "        print(reportlines[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import datetime\n",
    "currenttime = datetime.datetime.now()\n",
    "\n",
    "newfilename='genomicsreportfile'+'-'+str(currenttime.year)+'-'+str(currenttime.month)+'-'+str(currenttime.day)+'-'+str(currenttime.hour)\\\n",
    "    +'-'+str(currenttime.minute)+'-'+str(currenttime.second)+'-'+str(currenttime.microsecond)+'.txt'\n",
    "fileobject = open(newfilename, 'w')\n",
    "for i in range(len(reportlines)):\n",
    "    fileobject.write(str(reportlines[i]))\n",
    "    fileobject.write('\\n')\n",
    "fileobject.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from fpdf import FPDF \n",
    "import datetime \n",
    "pdf = FPDF()     \n",
    "pdf.add_page() \n",
    "pdf.set_font(\"Arial\", size = 8) \n",
    "currenttime = datetime.datetime.now()\n",
    "strpdffilename='genomicsreportfile'+'-'+str(currenttime.year)+'-'+str(currenttime.month)+'-'+str(currenttime.day)+'-'+str(currenttime.hour)\\\n",
    "    +'-'+str(currenttime.minute)+'-'+str(currenttime.second)+'-'+str(currenttime.microsecond)+'.pdf'\n",
    "\n",
    "for i in range(len(reportlines)):\n",
    "    #fileobject.write(str(reportlines[i]))\n",
    "    pdf.cell(200, 10, txt = str(reportlines[i]), ln = i+1, align = 'L')\n",
    "    \n",
    "# save the pdf with name .pdf \n",
    "pdf.output(strpdffilename)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
